v2/device.go

```diff
@@ -97,7 +98,11 @@ func New(opt *Options) *DeviceClient {
 	d.mqttOpt.OnConnectionLost = connectionLost
 	d.mqttOpt.OnConnect = onConnect
 	if opt.Will != nil {
-		d.mqttOpt.SetWill(opt.Will.Topic, opt.Will.Payload, opt.Qos, opt.Retain)
+		payload, err := json.Marshal(opt.Will.Payload)
```

Comment:
Sorry, I thought to do like https://github.com/eclipse/paho.mqtt.golang/blob/93c8a06e1e6dfeb17dd18a44e3e2f4e1db778237/client.go#L583-L591